### PR TITLE
Listen for requests on a Unix Domain Socket.

### PR DESCRIPTION
**Pull request by janzantinge**
Misk services should be able to listen on a Unix Domain Socket to enable efficient communication between an application and its sidecar (like Envoy).

In a service mesh setup, Misk services will not have to do TLS termination and will not have to worry about long lived connections, Envoy will take care of all of that. The reason for this change is to prevent a Misk service from listening for and responding to unencrypted traffic, instead letting the sidecar listen and proxy the traffic via the (not externally accessible) socket.

**Note**: this PR doesn't force egress traffic through the Unix Domain Socket. In some service mesh setups (like Istio), egress traffic is intercepted and forced through the sidecar in a way that's invisible to services (Misk or otherwise).

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt [hunk 1]
```diff
@@ -153,6 +154,19 @@ class JettyService @Inject internal constructor(
...
+    if (webConfig.unix_domain_socket != null) {
+      val udsConnector = UnixSocketConnector(
+        server,
+        null /* executor */,
+        null /* scheduler */,
+        null /* buffer pool */,
+        webConfig.selectors ?: -1,
+        listOf(HttpConnectionFactory(httpConfig)).toTypedArray()
+      )
+      udsConnector.setUnixSocket(webConfig.unix_domain_socket.path)
```
[the remark this answers is not in the document]
Comment (alecthomas) [reply misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: The two forms I'm aware of are `\0<name>` and `@<name>`.

[the remark this answers is not in the document]
Comment (janzantinge) [reply misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: the null byte barfed on me before, never occurred to me to try the '@' (which worked).

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt [hunk 2]
```diff
@@ -153,6 +154,19 @@ class JettyService @Inject internal constructor(
...
+      server.addConnector(udsConnector)
```
Comment (alecholmes) [inline misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: Right before this, you'll also want to register metrics. Here's how it's done with the other connectors:
```
      httpsConnector.addBean(connectionMetricsCollector.newConnectionListener(
          "https",
          webConfig.ssl.port
      ))
```

File: misk-testing/src/main/kotlin/misk/web/WebTestingModule.kt [hunk 3]
```diff
@@ -32,7 +32,11 @@ class WebTestingModule(
                 passphrase = "serverpassword",
                 format = SslLoader.FORMAT_PEM
             ),
-            mutual_auth = WebSslConfig.MutualAuth.NONE)
+            mutual_auth = WebSslConfig.MutualAuth.NONE
+        ),
+        unix_domain_socket = WebUnixDomainSocketConfig(
```
Comment (swankjesse) [inline misk-testing/src/main/kotlin/misk/web/WebTestingModule.kt]: dig

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt [hunk 4]
```diff
@@ -153,6 +154,19 @@ class JettyService @Inject internal constructor(
...
     }
 
+    if (webConfig.unix_domain_socket != null) {
+      val udsConnector = UnixSocketConnector(
+        server,
+        null /* executor */,
+        null /* scheduler */,
+        null /* buffer pool */,
+        webConfig.selectors ?: -1,
+        listOf(HttpConnectionFactory(httpConfig)).toTypedArray()
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: I think this is right.

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt (hunk 2 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: Yes. @janzantinge please follow-up!

**Pull request merged**